packages/roosterjs-content-model-core/lib/coreApi/setDOMSelection/setDOMSelection.ts

```diff
@@ -231,3 +241,23 @@ function setRangeSelection(doc: Document, element: HTMLElement | undefined, coll
         addRangeToSelection(doc, range, isReverted);
     }
 }
+
+function ensureImageHasSpanParent(image: HTMLImageElement) {
+    const parent = image.parentElement;
+
+    if (
+        parent &&
+        isNodeOfType(parent, 'ELEMENT_NODE') &&
+        isElementOfType(parent, 'span') &&
+        parent.firstElementChild == image &&
+        parent.lastElementChild == image &&
+        !parent.textContent
+    ) {
+        return parent;
+    }
+
+    const span = image.ownerDocument.createElement('span');
+    span.appendChild(image);
+    parent?.appendChild(span);
```

Comment:
this is not right. the IMAGE may not be the last element of parent, so this will change the order of image.

You can use existing util function `wrap`